lib/ts/recipe/thirdparty/api/implementation.ts

```diff
@@ -194,8 +195,17 @@ export default function getAPIInterface(): APIInterface {
         },
 
         appleRedirectHandlerPOST: async function ({ code, state, options }): Promise<void> {
+            const userContext = makeDefaultUserContextFromAPI(options.req);
+            const origin = await options.appInfo.origin(userContext);
```

Comment:
can't get origin from here cause this API is not called from the frontend, but called from apple. We need to instead use state param to get the origin. Undo the changes in this file - we will get back to this later on